exchanges/poloniex/poloniex_futures_types.go

```diff
@@ -0,0 +1,455 @@
+	target := [6]any{&v.OpenPrice, &v.HighPrice, &v.LowestPrice, &v.ClosingPrice, &v.StartTime, &v.EndTime}
+	return json.Unmarshal(data, &target)
+}
+
+// FuturesMarkPrice represents a mark price instance
+type FuturesMarkPrice struct {
+	MarkPrice types.Number `json:"mPx"`
+	Symbol    string       `json:"symbol"`
+	Timestamp types.Time   `json:"ts"`
+}
+
+// FuturesMarkPriceCandle represents a k-line data for mark price
+type FuturesMarkPriceCandle struct {
+	OpeningPrice types.Number
+	HighestPrice types.Number
+	LowestPrice  types.Number
+	ClosingPrice types.Number
+	StartTime    types.Time
+	EndTime      types.Time
+}
+
+// UnmarshalJSON deserializes byte data into FuturesMarkPriceCandle instance
+func (v *FuturesMarkPriceCandle) UnmarshalJSON(data []byte) error {
+	target := [6]any{&v.OpeningPrice, &v.HighestPrice, &v.LowestPrice, &v.ClosingPrice, &v.StartTime, &v.EndTime}
+	return json.Unmarshal(data, &target)
+}
+
+// ProductInfo represents basic information of the all product
+type ProductInfo struct {
+	Alias                 string       `json:"alias"`
+	BaseAsset             string       `json:"bAsset"`
+	BaseCurrency          string       `json:"bCcy"`
+	ContractType          string       `json:"ctType"`
+	ContractValue         types.Number `json:"ctVal"`
+	InitialMarginRate     types.Number `json:"iM"`
+	Leverage              types.Number `json:"lever"`
+	LotSize               types.Number `json:"lotSz"`
+	MaintenanceMarginRate types.Number `json:"mM"`
+	MaximumRiskLimit      types.Number `json:"mR"`
+	MaxLeverage           string       `json:"maxLever"`
+	MaxPrice              types.Number `json:"maxPx"`
+	MaxQuantity           types.Number `json:"maxQty"`
+	MinPrice              types.Number `json:"minPx"`
+	MinQuantity           types.Number `json:"minQty"`
+	MinSize               types.Number `json:"minSz"`
+	ListingDate           types.Time   `json:"oDate"`
+	PriceScale            string       `json:"pxScale"`
+	QuoteCurrency         string       `json:"qCcy"`
+	SettlementCurrency    string       `json:"sCcy"`
+	Status                string       `json:"status"`
+	Symbol                string       `json:"symbol"`
+	TickSize              types.Number `json:"tSz"`
+	TradableStartTime     types.Time   `json:"tradableStartTime"`
+	VisibleStartTime      types.Time   `json:"visibleStartTime"`
+	OrderPriceRange       types.Number `json:"ordPxRange"`
+	MarketMaxQty          types.Number `json:"marketMaxQty"`
+	LimitMaxQty           types.Number `json:"limitMaxQty"`
+}
+
+// FuturesFundingRate represents symbols funding rate information
+type FuturesFundingRate struct {
+	Symbol                   string       `json:"s"`
+	FundingRate              types.Number `json:"fR"`
+	FundingRateSettleTime    types.Time   `json:"fT"`
+	NextPredictedFundingRate types.Number `json:"nFR"`
+	NextFundingTime          types.Time   `json:"nFT"`
+	Timestamp                types.Time   `json:"ts"`
```

Comment:
Not seeing this field being returned?

Reply:
It's included in the websocket response.
<img width="582" height="417" alt="Screenshot 2025-10-27 at 10 39 51 in the morning" src="https://github.com/user-attachments/assets/8f29e12a-ebd5-473c-84c0-0340628dd224" />